gobblin-cluster/src/main/java/gobblin/cluster/GobblinHelixJobScheduler.java

```diff
@@ -174,6 +182,15 @@ public NonScheduledJobRunner(Properties jobConfig, JobListener jobListener) {
     public void run() {
       try {
         GobblinHelixJobScheduler.this.runJob(this.jobConfig, this.jobListener);
+
+        // remove non-scheduled job catalog once done so it won't be re-executed
+        if (GobblinHelixJobScheduler.this.jobCatalog != null) {
```

Comment:
so if the master goes down before the job completes: 
- would the job be run again once master comes back up? [1]
- will it be removed after execution in [1] ?